### DISCUSSION TITLE
Help me fix the old casadei code for the new version

### DISCUSSION THREAD

**Discussion by Imil657** [casadi/casadi#3742] [Q&A] [answered]
Hi everyone, I have recently started programming and would like to use the code from https://mintoc.de/index.php/Gravity_Turn_Maneuver_ (Casadi) follows the trajectory of the rocket takeoff, but it is written in Casadi 2.4.2, and my python does not launch it. I would be very grateful if you could help me fix the code for the Casadei 3.6.4 version, otherwise nothing has been working for six months.

**Comment by tmmsartor** [comment, marked as answer]
Have a look here: https://github.com/tmmsartor/casadi_issues/blob/main/3742_fix_old_casadi_gravity_turn/main.py,
this works with casadi 3.6.5. I also added some plot code to check the results.

**Reply by Imil657** [reply]
Wow, thanks. I didn't expect anyone to change the code, I thought they would just give me a couple of tips. Thank you very much
